app/cosi/README.md

```diff
@@ -0,0 +1,338 @@
+Then you can fetch, update, compile and install the cosi-binary using:
+
+```bash
+go get -u github.com/dedis/cothority/app/cosi
+```
+
+The `cosi` binary will be installed in the directory indicated by `$GOBIN`.
+
+# Command-line Interface
+
+The `cosi` application provides both a client for signing messages,
+and a server implementing the cosigner or witness-server role
+in the CoSi protocol.
+
+## Chose the Right Directory for the Examples
+
+For the examples in the following sections, we suppose you're in a directory
+where you can find the following files: `README.md` and `dedis-group.toml`.
+
+If you compiled from source, please change the directory like so:
+
+```bash
+cd $GOPATH/src/github.com/dedis/cothority/app/cosi/
+```
+
+If you used the binary distribution, please use
+
+```bash
+cd $( dirname $( which cosi ) )
+```
+
+## Collectively Signing Messages with the CoSi Client
+
+In order to sign messages collectively, you first need to define the set of
+cosigners that will participate.  To do this, you need to prepare a *group definition* 
+file which lists the cosigners to use with their public keys and Internet addresses.
+You may use [our default list of public CoSi
+servers](https://github.com/dedis/cosi/blob/master/dedis_group.toml) if you wish, or define your own.
+
+CoSi will by default search for a file "group.toml" in the default configuration folders
+which are `$HOME/.config/cosi/` for Linux systems and `$HOME/Library/cosi/` for
+mac systems. If CoSi did not find anything, the default mechanism is to search in the current
+directory.
+
+Once you have a valid group definition, you can sign a file using the `cosi sign’
+command.  Here is an example that uses our default public CoSi server group to
+sign the README.md file at the top of the cosi source directory:
+
+```bash
+cosi sign -g dedis_group.toml README.md
+```
+
+When collective signing completes,
+the resulting signature will be written to standard output by default.
+To write the signature written to a file,
+you may redirect output or use the the `-o` option:
+
+```base
+cosi sign -g dedis_group.toml -o README.sig README.md
+```
+
+To verify a collective signature, use the `cosi verify` command:
+  
+```bash
+cosi verify -g dedis_group.toml -s README.sig README.md
+```
+
+Verification can also take the signature from standard input:
+
+```bash
+cat README.sig | cosi verify -g dedis_group.toml README.md
+```
+
+In the current prototype, CoSi witness servers do not validate or check the 
+messages you propose in any way; they merely serve to provide transparency
+by publicly attesting the fact that they have observed and cosigned the message.
+A future CoSi release will add support for message validation plugins,
+by which the servers can apply application-specific checks to messages
+before signing off on them,
+e.g., to validate a [collectively signed blockchain](http://arxiv.org/abs/1602.06997).
+
+## Running Your Own CoSi Witness Server
```

Comment:
This section already exists, here: https://github.com/dedis/cothority/blob/master/app/cothorityd/README.md